src/react/context/ApolloProvider.tsx

```diff
@@ -18,29 +18,28 @@ export const ApolloProvider: React.FC<ApolloProviderProps<any>> = ({
   children
 }) => {
   const ApolloContext = getApolloContext();
-  return (
-    <ApolloContext.Consumer>
-      {(context: any = {}) => {
-        if (client && context.client !== client) {
-          context = Object.assign({}, context, { client });
-        }
+  const parentContext = React.useContext(ApolloContext);
 
-        if (suspenseCache) {
-          context = Object.assign({}, context, { suspenseCache });
-        }
+  const newContext = React.useMemo(() => {
+    let context = parentContext;
+    if (client && parentContext.client !== client) {
+      context = Object.assign({}, context, { client });
+    }
+    if (suspenseCache && parentContext.suspenseCache !== suspenseCache) {
+      context = Object.assign({}, context, { suspenseCache });
+    }
+    return context;
```

Comment:
[nit] Since this is now wrapped in a `useMemo`, I think we can simplify this a bit: 

```suggestion
    return {
      ...parentContext,
      client: client || parentContext.client,
      suspenseCache: suspenseCache || parentContext.suspenseCache
    }
```

Since we know `useMemo` won't run again unless `parentContext`, `client`, or `suspenseCache` changes, we can simplify by always returning a new object here since that is expected anyways. Since we always prefer the passed in `client` and/or `suspenseCache`, we can use it if its passed, otherwise fallback to the parent value.

Reply:
No idea why this comment got added twice. Respond to the other one.